homework/assignment_5/assignment_5.py

```diff
@@ -0,0 +1,458 @@
+            temp_df["population_proportion"]
+        )  # wmse
+
+        # plots for each predictor
+        temp_df["bin"] = temp_df["bin"].astype("str")  # need to convert to str to plot
+        binplot = px.bar(temp_df, x="bin", y="bin_count", title=column)
+        binplot.write_html(f"graphs/{column}_binplot.html")
+
+        # getting links
+        joiner.append(f"{column}")
+        names.append(f"{column}_binplot")
+        urls.append(f"graphs/{column}_binplot.html")
+
+        pred_name = train_cont[column].name
+        l1.append(pred_name)
+        l2.append(temp_df["diff_mean_resp"].mean())
+        l3.append(temp_df["weighted_diff"].mean())
+
+    # predictors and mse, wmse
+    mse_df["predictor"] = l1
+    mse_df["mse"] = l2
+    mse_df["weighted_mse"] = l3
+
+    # making graphs df to merge with actual
+    graphs_df["joiner"] = joiner
+    graphs_df["name"] = names
+    graphs_df["url"] = urls
+    graphs_df["name_url"] = graphs_df["name"] + "#" + graphs_df["url"]
+
+    # joining brute force and mse dataframes
+    # merging on pred 1
+    brute_force = brute_force.merge(mse_df, left_on="Predictor 1", right_on="predictor")
+    brute_force = brute_force.drop(["predictor"], axis=1)
+    brute_force = brute_force.rename(
+        columns={"mse": "mse_1", "weighted_mse": "weighted_mse_1"}
+    )
+
+    # merging on pred 2
+    brute_force = brute_force.merge(mse_df, left_on="Predictor 2", right_on="predictor")
+    brute_force = brute_force.drop(["predictor"], axis=1)
+    brute_force = brute_force.rename(
+        columns={"mse": "mse_2", "weighted_mse": "weighted_mse_2"}
+    )
+
+    # calculating actual mse, wmse
+    brute_force["Difference of Mean Response"] = (
+        brute_force["mse_1"] * brute_force["mse_2"]
+    )
+    brute_force["Weighted Difference of Mean Response"] = (
+        brute_force["weighted_mse_1"] * brute_force["weighted_mse_2"]
+    )
+    brute_force = brute_force.drop(
+        ["mse_1", "mse_2", "weighted_mse_1", "weighted_mse_2"], axis=1
+    )
+
+    # correlation of predictors
+    brute_force["Pearson"] = pearsons_r
+    brute_force["Absolute Pearson"] = abs_pearson
+
+    # merging graphs_df on actual to link graphs
+    brute_force = brute_force.merge(graphs_df, left_on="Predictor 1", right_on="joiner")
+    brute_force = brute_force.drop(columns={"joiner", "name", "url"})
+    brute_force = brute_force.rename(columns={"name_url": "Predictor 1 Bin Plot"})
+    brute_force = brute_force.merge(graphs_df, left_on="Predictor 2", right_on="joiner")
+    brute_force = brute_force.drop(columns={"joiner", "name", "url"})
+    brute_force = brute_force.rename(columns={"name_url": "Predictor 2 Bin Plot"})
+
+    ##################################################
+    # MODELS
+    ##################################################
+
+    # Decision tree
+    clf = tree.DecisionTreeClassifier()
+    clf = clf.fit(X_train, y_train)
+    tree.plot_tree(clf)
```

Comment:
Getting a weird error here I think has to do with graph viz, I ended up downloading it to my system and then it worked.